addons/sourcemod/scripting/nd_structure_intercept.sp

```diff
@@ -0,0 +1,212 @@
+#include <sourcemod>
+#include <dhooks>
+#include <sdkhooks>
+
+#tryinclude <nd_commander_build>
+
+#if !defined _nd_commmander_build_included_
+
+    enum eNDStructures
+    {
+        eND_CommandBunker = 0,
+        eND_MachineGunTurret,
+        eND_TransportGate,
+        eND_PowerStation,
+        eND_WirelessRepeater,
+        eND_RelayTower,
+        eND_SupplyStation,
+        eND_Assembler,
+        eND_Armory,
+        eND_Artillery,
+        eND_RadarStation,
+        eND_FlamethrowerTurret,
+        eND_SonicTurret,
+        eND_RocketTurret,
+        eND_Wall,
+        eND_Barrier
+    }
+
+    /**
+     * Allows a plugin to block a structure form being built by returning Plugin_Stop
+     * or change the structure and/or position by returning Plugin_Changed.
+     *
+     * @param int client                     client index of the commander who attempted to build the structure
+     * @param eNDStructures &structure       type of structure being built
+     * @param float position[3]              x,y,z coordinate of where structure is being built
+     * @return                               Action that should be taken
+     */
+    forward Action ND_OnCommanderBuildStructure(int client, eNDStructures &structure, float position[3]);
```

Comment:
We don't need to include this forward because it's not directly used inside the calling plugin.

Reply:
Addressed in https://github.com/stickz/Redstone/commit/4a806ac48ffbdbd305ccc1895f9e8f4925e23b15